Source/JavaScriptCore/assembler/MacroAssemblerX86Common.h

```diff
@@ -2494,6 +2494,16 @@ class MacroAssemblerX86Common : public AbstractMacroAssembler<Assembler> {
     //
     // Move values in registers.
 
+    // Move and zeroes out the upper 32-bits.
+    void move32(RegisterID src, RegisterID dest)
+    {
+#if CPU(X86_64)
+        m_assembler.movl_rr(src, dest);
+#else
+        move(src, dest);
```

Comment:
I believe the X86 Macro Assembler elides the move if src == dest.

Reply:
That is the intended behavior.  There's no upper 32-bit to worry about for 32-bit registers.

Reply:
You are correct.